win32/src/host.rs

```diff
@@ -39,14 +39,73 @@ pub trait Window {
 }
 
 #[cfg_attr(feature = "wasm", wasm_bindgen::prelude::wasm_bindgen)]
-pub enum FileAccess {
-    READ,
-    WRITE,
+#[derive(Debug, Clone)]
+pub struct FileOptions {
+    /// Permit read access.
+    pub read: bool,
+    /// Permit write access.
+    pub write: bool,
+    /// Truncate the file to zero length.
+    pub truncate: bool,
+    /// Create the file if it doesn't exist.
+    pub create: bool,
+    /// Create the file if it doesn't exist, and fail if it does.
+    pub create_new: bool,
+}
+
+impl FileOptions {
+    pub fn empty() -> Self {
+        Self {
+            read: false,
+            write: false,
+            truncate: false,
+            create: false,
+            create_new: false,
+        }
+    }
+
+    pub fn read() -> Self {
+        Self {
+            read: true,
+            write: false,
+            truncate: false,
+            create: false,
+            create_new: false,
+        }
+    }
 }
 
 pub trait File: std::io::Read + std::io::Write + std::io::Seek {
-    /// Just file size for now, but maybe we'll need more(?)
-    fn info(&self) -> u32;
+    fn stat(&self) -> Result<Stat, u32>;
+}
+
+pub trait FindHandle {
+    fn next(&mut self) -> Result<Option<FindFile>, u32>;
+}
+
+#[derive(Debug, Clone)]
+pub struct FindFile {
+    pub path: String,
+    pub name: String,
```

Comment:
Are these the containing dir and filename?  (not familiar with the API it's exposing here)

Similar to above I wonder if it's better to design a narrower API here and hide more of this find-next stuff within the win32 lib.  Just thinking about implementing these types for each of Windows/Unix/web makes me wonder.

Reply:
Yes, it’s full path and filename. But the win32 side could extract filename using some helper and we could avoid duplicating it here.

I agree that this host.find API isn’t the best at the moment. The problem is that we need to wrap ReadDir to lazily iterate through dir contents, so it requires a separate type regardless. I’ll see what I can do to improve.